apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/EventActivity.java

```diff
@@ -123,22 +127,58 @@ public void toggleTransmissionEnabled(View view) {
         }
     }
 
+    public void toggleDeviceIdEnabled(View view) {
+        final AnalyticsTransmissionTarget target = getSelectedTarget();
+        if (target != null) {
+            updateButtonStates(target);
+
+            // TODO remove reflection once new APIs available in jCenter.
+            // target.getPropertyConfigurator().collectDeviceId();
+            {
+                Method method;
+                try {
+                    method = PropertyConfigurator.class.getMethod("collectDeviceId");
+                } catch (Exception ignored) {
+                    return;
+                }
+                try {
+                    method.invoke(target.getPropertyConfigurator());
+                } catch (IllegalAccessException ignored) {
+                } catch (InvocationTargetException ignored) {
+                }
+            }
+            mDeviceIdEnabledCheckBox.setChecked(true);
+            mDeviceIdEnabledCheckBox.setText(R.string.device_id_enabled);
+            mDeviceIdEnabledCheckBox.setEnabled(false);
+        }
+    }
+
     private void updateButtonStates(AnalyticsTransmissionTarget target) {
         if (target == null) {
             mTransmissionEnabledCheckBox.setVisibility(View.GONE);
             mConfigureTargetPropertiesButton.setVisibility(View.GONE);
             mOverrideCommonSchemaButton.setVisibility(View.GONE);
+            mDeviceIdEnabledCheckBox.setVisibility(View.GONE);
         } else {
             mTransmissionEnabledCheckBox.setVisibility(View.VISIBLE);
             mConfigureTargetPropertiesButton.setVisibility(View.VISIBLE);
             mOverrideCommonSchemaButton.setVisibility(View.VISIBLE);
             boolean enabled = target.isEnabledAsync().get();
             mTransmissionEnabledCheckBox.setChecked(enabled);
-            if (enabled) {
-                mTransmissionEnabledCheckBox.setText(R.string.transmission_enabled);
-            } else {
-                mTransmissionEnabledCheckBox.setText(R.string.transmission_disabled);
+            mTransmissionEnabledCheckBox.setText(enabled ? R.string.transmission_enabled : R.string.transmission_disabled);
+
+            // TODO remove reflection once new APIs available in jCenter.
+            {
+                try {
+                    PropertyConfigurator.class.getMethod("collectDeviceId");
+                } catch (Exception ignored) {
+                    return;
+                }
             }
+            mDeviceIdEnabledCheckBox.setVisibility(View.VISIBLE);
```

Comment:
It shouldn't reset settings to disabled. `collectDeviceId` can only be called once before `start`, there is no disable or reset device Id functionalities.

Reply:
This isn't preventing from unchecking. It is enabling the checkbox.
If you enable device on a transmission target and come back to the target after switching to another target, it will be "check-able" which shouldn't be.